lego/apps/users/views/users.py

```diff
@@ -173,3 +174,25 @@ def change_grade(self, request, *args, **kwargs):
                 newGrade.add_user(user)
 
         return Response(MeSerializer(user).data)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        permission_classes=[IsAuthenticated],
+        serializer_class=PhotoConsentSerializer,
+    )
+    def update_photo_consent(self, request, *args, **kwargs):
+        if not request.user.has_perm(CREATE, PhotoConsent):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = self.get_object()
+
+        with transaction.atomic():
+            photo_consent = user.photo_consents.get(
+                semester=request.data.get("semester"), domain=request.data.get("domain")
+            )
```

Comment:
What happens when the photo_consent does not exist?

Reply:
There has to be photo consents in order for the user to send a request to change them. (front end) However, I added validation and creation of a photo consent here in the wierd case this is not generated beforehand. :shrug: